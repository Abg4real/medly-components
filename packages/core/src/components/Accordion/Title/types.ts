import { WithThemeProp } from '@medly-components/utils';

export interface Props extends WithThemeProp{
    color?: string;
    isActive?: boolean;
    onToggle?: () => void;
    bgColor?: string;

}


