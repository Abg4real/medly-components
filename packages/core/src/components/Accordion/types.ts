import { Props as TitleProps } from './Title';

export interface Props {
    Title?: React.SFC<TitleProps>;
    Content?: React.SFC;
}