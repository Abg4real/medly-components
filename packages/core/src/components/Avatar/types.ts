import { AvatarSizes } from '@medly-components/theme';
import { WithThemeProp } from '@medly-components/utils';

export interface Props extends WithThemeProp {
    size?: keyof AvatarSizes;
    textColor?: string;
    bgColor?: string;
}

export interface StyledProps extends Props {
    fontSize: string;
    avatarSize: string;
}
