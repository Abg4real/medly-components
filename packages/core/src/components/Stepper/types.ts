import { StepperSizes } from '@medly-components/theme';
import { HTMLProps, Omit, WithThemeProp } from '@medly-components/utils';

export interface CounterProps extends WithThemeProp, Omit<HTMLProps<HTMLDivElement>, 'size'> {
    size?: StepperSizes;
    active?: boolean;
}

export interface CounterStyledProps extends CounterProps {
    counterSize: string;
    fontSize: string;
    bgColor: string;
    descriptionColor: string;
}

export interface StepProps extends CounterProps {
    step?: number;
}

export interface Props {
    /** Stepper Size */
    size?: StepperSizes;
    /** Active step */
    activeStep?: number;
    /** Function to be called on clicking on any step */
    onStepClick?: (step: number) => void;
}

export interface StepperStaticProps {
    Step: React.SFC<StepProps>;
}
