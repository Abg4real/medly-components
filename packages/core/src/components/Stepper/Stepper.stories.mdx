import { Stepper } from './Stepper';
import { defaultTheme } from '@medly-components/theme';
import { Preview, Story, Meta, Props } from '@storybook/addon-docs/blocks';
import * as stories from './Stepper.stories';

<Meta title="Core/Stepper" component={Stepper} />

# Stepper

### Basic Stepper

Use `Stepper.Step` to add step in `Stepper` component. Like bellow . . .

```typescript
<Stepper activeStep={0} size="S" onStepClick={setActiveStep}>
    <Stepper.Step>Customer Information</Stepper.Step>
    <Stepper.Step>Payment information</Stepper.Step>
</Stepper>
```

<Preview withToolbar>
    <Story name="Basic">{stories.Basic}</Story>
</Preview>

### Props

<Props of={Stepper} />

### StepperTheme

<Props of={stories.ThemeInterface} />

### StepperTheme Defaults

<code>{JSON.stringify(defaultTheme.stepper, null, 2)}</code>
