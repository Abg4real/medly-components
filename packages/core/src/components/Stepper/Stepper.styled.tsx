import { styled } from '@medly-components/utils';

export const StepperStyled = styled('div')`
    display: flex;
`;
