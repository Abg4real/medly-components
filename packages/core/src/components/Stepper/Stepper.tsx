import { WithStyle } from '@medly-components/utils';
import React from 'react';
import Step from './Step/Step';
import { StepperStyled } from './Stepper.styled';
import { Props, StepperStaticProps } from './types';

const Stepper: React.SFC<Props> & WithStyle & StepperStaticProps = props => {
    const { size, activeStep, children, onStepClick } = props;
    return (
        <StepperStyled>
            {React.Children.map(children, (child, index) => {
                return React.cloneElement(child as any, {
                    size,
                    key: index + 1,
                    step: index + 1,
                    active: activeStep === index,
                    onClick: onStepClick ? () => onStepClick(index) : undefined
                });
            })}
        </StepperStyled>
    );
};

Stepper.defaultProps = {
    activeStep: 0
};
Stepper.Step = Step;
Stepper.Style = StepperStyled;
Stepper.displayName = 'Stepper';

export default Stepper;
