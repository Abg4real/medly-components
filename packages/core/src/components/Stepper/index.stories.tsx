import { select } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React, { useState } from 'react';
import { Stepper } from './Stepper';
import { Props } from './types';

const size: Array<Props['size']> = ['S', 'M', 'L', 'XL'];

const DemoComponent = () => {
    const [activeStep, setActiveStep] = useState(0);
    return (
        <Stepper activeStep={activeStep} size={select('Size', size, 'S')} onStepClick={setActiveStep}>
            <Stepper.Step>Customer Information</Stepper.Step>
            <Stepper.Step>Payment information</Stepper.Step>
            <Stepper.Step>Booking is confirmed!</Stepper.Step>
        </Stepper>
    );
};
storiesOf('Core', module).add('Stepper', DemoComponent, {
    props: {
        propTablesExclude: [Stepper.Step]
    }
});
