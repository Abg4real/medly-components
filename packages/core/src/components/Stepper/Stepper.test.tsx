import { TestUtils } from '@medly-components/utils';
import React from 'react';
import { Stepper } from './Stepper';
import { Props } from './types';

const renderer = (props: Props = {}) =>
    TestUtils.render(
        <Stepper {...props}>
            <Stepper.Step>Customer Information</Stepper.Step>
            <Stepper.Step>Payment information</Stepper.Step>
            <Stepper.Step>Booking is confirmed!</Stepper.Step>
        </Stepper>
    );

describe('Stepper component', () => {
    it('should render correctly with all the default props', () => {
        const { container } = renderer();
        expect(container).toMatchSnapshot();
    });

    it('should render correctly with all the props given', () => {
        const mockOnStepClick = jest.fn();
        const { container } = renderer({ activeStep: 0, size: 'S', onStepClick: mockOnStepClick });
        expect(container).toMatchSnapshot();
    });

    it('should change active step when click on any step', async () => {
        const mockOnStepClick = jest.fn();
        const { container, getByText } = renderer({ activeStep: 0, size: 'S', onStepClick: mockOnStepClick });
        TestUtils.fireEvent.click(getByText('Payment information'));
        expect(mockOnStepClick).toBeCalledWith(1);
        expect(container).toMatchSnapshot();
    });
});
