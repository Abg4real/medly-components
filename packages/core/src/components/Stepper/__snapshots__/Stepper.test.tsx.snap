// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper component should change active step when click on any step 1`] = `
.c5 {
  font-size: 14px;
  font-weight: 700;
  color: inherit;
  text-transform: uppercase;
}

.c6 {
  font-size: 14px;
  font-weight: 700;
  color: inherit;
}

.c3 {
  display: inline-block;
  text-align: center;
  width: 30px;
  height: 30px;
  line-height: 30px;
  border-radius: 50%;
  overflow: hidden;
  color: #ffffff;
  background: #012040;
}

.c3 .c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: 16px;
}

.c3 + .c4 {
  color: #012040;
  font-size: 16px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c7 {
  display: inline-block;
  text-align: center;
  width: 30px;
  height: 30px;
  line-height: 30px;
  border-radius: 50%;
  overflow: hidden;
  color: #ffffff;
  background: #C7D0D8;
}

.c7 .c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: 16px;
}

.c7 + .c4 {
  color: #C7D0D8;
  font-size: 16px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 20px;
}

.c1 .c2 {
  margin-right: 10px;
}

.c1 * {
  cursor: pointer;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2 c3"
        font-size="16px"
      >
        <strong
          class="c4 c5"
        >
          1
        </strong>
      </div>
      <strong
        class="c4 c6"
      >
        Customer Information
      </strong>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2 c7"
        font-size="16px"
      >
        <strong
          class="c4 c5"
        >
          2
        </strong>
      </div>
      <strong
        class="c4 c6"
      >
        Payment information
      </strong>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2 c7"
        font-size="16px"
      >
        <strong
          class="c4 c5"
        >
          3
        </strong>
      </div>
      <strong
        class="c4 c6"
      >
        Booking is confirmed!
      </strong>
    </div>
  </div>
</div>
`;

exports[`Stepper component should render correctly with all the default props 1`] = `
.c5 {
  font-size: 14px;
  font-weight: 700;
  color: inherit;
  text-transform: uppercase;
}

.c6 {
  font-size: 14px;
  font-weight: 700;
  color: inherit;
}

.c3 {
  display: inline-block;
  text-align: center;
  width: 30px;
  height: 30px;
  line-height: 30px;
  border-radius: 50%;
  overflow: hidden;
  color: #ffffff;
  background: #012040;
}

.c3 .c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: 16px;
}

.c3 + .c4 {
  color: #012040;
  font-size: 16px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c7 {
  display: inline-block;
  text-align: center;
  width: 30px;
  height: 30px;
  line-height: 30px;
  border-radius: 50%;
  overflow: hidden;
  color: #ffffff;
  background: #C7D0D8;
}

.c7 .c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: 16px;
}

.c7 + .c4 {
  color: #C7D0D8;
  font-size: 16px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 20px;
}

.c1 .c2 {
  margin-right: 10px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2 c3"
        font-size="16px"
      >
        <strong
          class="c4 c5"
        >
          1
        </strong>
      </div>
      <strong
        class="c4 c6"
      >
        Customer Information
      </strong>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2 c7"
        font-size="16px"
      >
        <strong
          class="c4 c5"
        >
          2
        </strong>
      </div>
      <strong
        class="c4 c6"
      >
        Payment information
      </strong>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2 c7"
        font-size="16px"
      >
        <strong
          class="c4 c5"
        >
          3
        </strong>
      </div>
      <strong
        class="c4 c6"
      >
        Booking is confirmed!
      </strong>
    </div>
  </div>
</div>
`;

exports[`Stepper component should render correctly with all the props given 1`] = `
.c5 {
  font-size: 14px;
  font-weight: 700;
  color: inherit;
  text-transform: uppercase;
}

.c6 {
  font-size: 14px;
  font-weight: 700;
  color: inherit;
}

.c3 {
  display: inline-block;
  text-align: center;
  width: 30px;
  height: 30px;
  line-height: 30px;
  border-radius: 50%;
  overflow: hidden;
  color: #ffffff;
  background: #012040;
}

.c3 .c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: 16px;
}

.c3 + .c4 {
  color: #012040;
  font-size: 16px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c7 {
  display: inline-block;
  text-align: center;
  width: 30px;
  height: 30px;
  line-height: 30px;
  border-radius: 50%;
  overflow: hidden;
  color: #ffffff;
  background: #C7D0D8;
}

.c7 .c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: 16px;
}

.c7 + .c4 {
  color: #C7D0D8;
  font-size: 16px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 20px;
}

.c1 .c2 {
  margin-right: 10px;
}

.c1 * {
  cursor: pointer;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2 c3"
        font-size="16px"
      >
        <strong
          class="c4 c5"
        >
          1
        </strong>
      </div>
      <strong
        class="c4 c6"
      >
        Customer Information
      </strong>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2 c7"
        font-size="16px"
      >
        <strong
          class="c4 c5"
        >
          2
        </strong>
      </div>
      <strong
        class="c4 c6"
      >
        Payment information
      </strong>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2 c7"
        font-size="16px"
      >
        <strong
          class="c4 c5"
        >
          3
        </strong>
      </div>
      <strong
        class="c4 c6"
      >
        Booking is confirmed!
      </strong>
    </div>
  </div>
</div>
`;
